### DISCUSSION TITLE
Randomly Stops!

### DISCUSSION THREAD

**Discussion by karlhfr** [winder/Universal-G-Code-Sender#1972] [Help]
Hi, can anyone help?

Getting random crashes. using a 3dtek cnc machine and when running a design to be cut, the machine just stops.

the time still continues but the router is in the same place. it can happen at different points of the cut. 

we have to disconnect, and re set up the 0's and start again. it can then get past the point it stopped and stop again at a different stage.

there doesn't seem to be any errors on the screen. 

we are very new to all this so any help would be greatly appreciated.

using a macbookbook air, with UGS version 2.0.11

**Comment by ElsGear** [comment]
What CNCcontroller do you use?
With my Smoothieboard on Windows7+10 and UGS version 2.0.11+12, I have the same problem for XY only but the Z-stepper is proceeding with the gcode.
Does your Z-stepper work? I had to change to another gcode with many Z-movement to visualize it.

I've tried the Smoothieboard on Pronterface for debugging purpose only and worked great.
If I can't figure this out I have to switch to bCNC or Smoopi to see what's wrong.
